src/sentry/api/serializers/models/grouphash.py

```diff
@@ -12,19 +12,19 @@ def get_latest_events(group_hash_list):
     Returns a list of events (or ``None``) in the same order as the input
     sequence.
     """
-    group_hashes_by_project_id = defaultdict(list)
+    group_hashes_by_group_id = defaultdict(list)
     for group_hash in group_hash_list:
-        group_hashes_by_project_id[group_hash.project_id].append(group_hash)
+        group_hashes_by_group_id[group_hash.group_id].append(group_hash)
 
     events_by_group_hash = {}
-    for project_id, group_hash_list_chunk in group_hashes_by_project_id.items():
+    for group_id, group_hash_list_chunk in group_hashes_by_group_id.items():
         event_id_list = GroupHash.fetch_last_processed_event_id(
-            project_id, [i.id for i in group_hash_list_chunk]
+            group_id, [i.id for i in group_hash_list_chunk]
         )
         event_by_event_id = {
             event.event_id: event
             for event in Event.objects.filter(
```

Comment:
This doesn't actually use the correct index, I'm going to fix this to use the btree on `(project_id, message_id)`.

Reply:
Fixed this in 6e0a627

Reply:
Not sure what I was on about here, this is using the `event.id` primary key and not `event.event_id`…